Assets/Scripts/Dialogue System/RunTime/DialogueHandler.cs

```diff
@@ -0,0 +1,345 @@
+using UnityEngine;
+using TMPro;
+using UnityEngine.UI;
+using System.Linq;
+using UnityEngine.EventSystems;
+using System;
+using UnityEngine.Events;
+using UnityEngine.InputSystem;
+using static UnityEngine.InputSystem.InputAction;
+
+public class DialogueHandler : MonoBehaviour
+{
+    [SerializeField] private DialogueContainer dialogueContainer;
+    [SerializeField] private TMP_Text currentDialogueText;
+    [SerializeField] private GameObject buttonPrefab;
+    [SerializeField] private Vector2 firstButtonPosition;
+    [SerializeField] private Vector2 buttonDisplacement;
+    [SerializeField] private GameObject parentCanvas;
+
+    [SerializeField]
+    [Tooltip("The keybinds that allow the player to move forward in the dialogue (when not choosing a specific response).")]
+    InputAction InteractAction;
+
+    [Tooltip("Text should have the first character's name")]
+    [SerializeField] private TMP_Text currentCharacterSpeakingNameText;
+
+    [Tooltip("GameObjects with Component that implements IChecker (Index order must match indices in Graph)")]
+    [SerializeField] private GameObject[] checkers;
+
+    [SerializeField] private UnityEvent[] events;
+
+    /// <summary>
+    /// Event is triggered when a new character node is reached.
+    /// The string inside the port of the node (which is assumed to be the new character's name) is passed in.
+    /// </summary>
+    public event Action<string> NewCharacterName;
+
+    public event Action EndDialogueNodeReached;
+
+    private const int checkerSubStrEndIndex = 6;
+    private const string checkerSubStrPass = "Pass: ";
+    private const string checkerSubStrFail = "Fail: ";
+    private const int timerSubStrEndIndex = 12;
+    private const string timerSubStr = "Time (sec): ";
+
+    //private NodeLinkData currentNodeLinkData;
+    private DialogueNodeData currentDialogueNodeData;
+
+    private List<GameObject> currentButtons = new List<GameObject>();
+
+    private string moveForwardToPortID = null;
+    private bool moveForwardOnInput = false;
+    private float timeToStayOnCurrentNode;
+    private bool useTimer = false;
+    private float timeEnteredCurrentNode;
+
+    private void OnEnable()
+    {
+        InteractAction.performed += OnMoveForward;
+        InteractAction.Enable();
+    }
+
+    private void OnDisable()
+    {
+        InteractAction.performed -= OnMoveForward;
+        InteractAction.Disable();
+    }
+
+    void Awake()
+    {
+        //Get the first node link, which has the PortName "Next"
+        NodeLinkData firstNodeLink = this.dialogueContainer.NodeLinks.First(node => node.PortName == "Next");
+            
+        //Get the first dialogue node by following where the link goes
+        this.currentDialogueNodeData = this.dialogueContainer.DialogueNodeData.First(node => node.NodeGuID == firstNodeLink.TargetNodeGuID);
+
+        HandleSpecialNodes();
+        GetCurrentScreen();
+    }
+
+    private void Update()
+    {
+        if (this.useTimer)
```

Comment:
make this into a guard clause to avoid indenting more than we need to, to make the code more readable
rather than:
```cs
if (a) {
  ...
}
```
do
```cs
if (!a) {
  return;
}
...
```